dependency_list_export.go

```diff
@@ -0,0 +1,107 @@
+
+type DependencyListExportService struct {
+	client *Client
+}
+
+// createDependencyListExportOptions represents the available CreateDependencyListExport()
+// options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/dependency_list_export.html#create-a-pipeline-level-dependency-list-export
+type createDependencyListExportOptions struct {
+	ExportType string `url:"export_type" json:"export_type"`
+}
+
+// DependencyListExport represents a request for a GitLab project's dependency list.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/dependency_list_export.html#create-a-pipeline-level-dependency-list-export
+type DependencyListExport struct {
+	ID          int    `json:"id"`
+	HasFinished bool   `json:"has_finished"`
+	Self        string `json:"self"`
+	Download    string `json:"download"`
+}
+
+// CreateDependencyListExport creates a new CycloneDX JSON export for all the project dependencies
+// detected in a pipeline.
+//
+// If an authenticated user does not have permission to read_dependency, this request returns a 403
+// Forbidden status code.
+//
+// SBOM exports can be only accessed by the export’s author.
+//
+// GitLab docs:
+// https://docs.gitlab.com/ee/api/dependency_list_export.html#create-a-pipeline-level-dependency-list-export
+func (s *DependencyListExportService) CreateDependencyListExport(id int, options ...RequestOptionFunc) (*DependencyListExport, *Response, error) {
+	// POST /pipelines/:id/dependency_list_exports
+	createExportPath := fmt.Sprintf("pipelines/%d/dependency_list_exports", id)
+
+	opt := &createDependencyListExportOptions{
+		ExportType: *Ptr("sbom"),
+	}
+
+	req, err := s.client.NewRequest(http.MethodPost, createExportPath, opt, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	export := new(DependencyListExport)
+	resp, err := s.client.Do(req, &export)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return export, resp, nil
+}
+
+// GetDependencyListExport gets metadata about a single dependency list export.
+//
+// GitLab docs:
+// https://docs.gitlab.com/ee/api/dependency_list_export.html#get-single-dependency-list-export
+func (s *DependencyListExportService) GetDependencyListExport(id int, options ...RequestOptionFunc) (*DependencyListExport, *Response, error) {
+	// GET /dependency_list_exports/:id
+	getExportPath := fmt.Sprintf("dependency_list_exports/%d", id)
+
+	req, err := s.client.NewRequest(http.MethodGet, getExportPath, nil, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	export := new(DependencyListExport)
+	resp, err := s.client.Do(req, &export)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return export, resp, nil
+}
+
+// DownloadDependencyListExport downloads a single dependency list export.
+//
+// GitLab docs:
+// https://docs.gitlab.com/ee/api/dependency_list_export.html#download-dependency-list-export
+func (s *DependencyListExportService) DownloadDependencyListExport(id int, options ...RequestOptionFunc) (string, *Response, error) {
```

Comment:
This seems to be one of the rare cases where it probably doesn't make sense to introduce a type for the return value. However, let's return a byte slice instead of a string.

And what do you think about pointing out `github.com/CycloneDX/cyclonedx-go` as a potential way to consume the returned `[]byte` ?

Seems to be fairly simple:

```go
bom := new(cdx.BOM)
decoder := cdx.NewBOMDecoder(reader, cdx.BOMFileFormatJSON)
if err = decoder.Decode(bom); err != nil {
	panic(err)
}
```

WDYT?

Reply:
I did the change to return a byte slice.

What would be the preferred way to suggest using `github.com/CycloneDX/cyclonedx-go` to parse the return value?
Would a function comment like this work?
```go
// DownloadDependencyListExport downloads a single dependency list export.
//
// The github.com/CycloneDX/cyclonedx-go package can be used to parse the returned bytes.
//
// GitLab docs:
// https://docs.gitlab.com/ee/api/dependency_list_export.html#download-dependency-list-export
func (s *DependencyListExportService) DownloadDependencyListExport...
```

Reply:
Yes, I've imagined a function comment. The idea behind this is that a user may be surprised and ask themselves what to do if we just return a byte slice.